### DISCUSSION TITLE
How to utilize the Replica in a Laravel application?

### DISCUSSION THREAD

**Discussion by jericopulvera** [planetscale/discussion#512] [Q&A] [answered]
According to the Laravel documentation (https://laravel.com/docs/10.x/database#read-and-write-connections), I need to determine the database host of the replica. Where can I find this information, and what will happen in the event of a failover? Will the database host switch between the two?

**Comment by mscoutermarsh** [comment, marked as answer]
Instructions for how to query a replica are available here: https://planetscale.com/docs/concepts/replicas

**Comment by toman** [comment]
You might look at https://laravel.com/docs/10.x/database#read-and-write-connections . Notice that while commonly only DB_HOST_READER and DB_HOST_WRITER are different, the text suggests that you can separate any of the configuration attributes in that array into values for read and write, including the database. Someone from Customer support at PlanetScale has pointed out to me that the database name <name>@replica works just as well as the USE @replica solution, and that seems to work.

**Comment by Rahul-codoffer** [comment]
You can configure in your database.php config file 
```
'connections' => [
    'mysql' => [
        'driver' => 'mysql',
        'host' => env('DB_HOST', ''),
        'port' => env('DB_PORT', '3306'),
        'database' => env('DB_DATABASE', 'your-database-name'),

        'read' => [
            'username' => env('DB_USERNAME_READ', 'replica-username'),
            'password' => env('DB_PASSWORD_READ', 'replica-password'),
        ],
        'write' => [
            'username' => env('DB_USERNAME_WRITE', 'primary-username'),
            'password' => env('DB_PASSWORD_WRITE', 'primary-password'),
        ],
        'charset' => 'utf8mb4',
        'collation' => 'utf8mb4_unicode_ci',
        'prefix' => '',
        'strict' => true,
        'sticky' => true,
        'engine' => null,
    ],
],
```
